pandas-stubs/tseries/holiday.pyi

```diff
@@ -1,3 +1,148 @@
+        ) = ...,
+        days_of_week: tuple[int, ...] | None = ...,
+    ) -> None: ...
+    @overload
+    def __init__(
+        self,
+        name: str,
+        year: int | None = ...,
+        month: int | None = ...,
+        day: int | None = ...,
+        offset: None = ...,
+        observance: Callable[[datetime], datetime] = ...,
+        # Values accepted by Timestamp(), or None:
+        start_date: (
+            np.integer | float | str | _date | datetime | np.datetime64 | None
+        ) = ...,
+        end_date: (
+            np.integer | float | str | _date | datetime | np.datetime64 | None
+        ) = ...,
+        days_of_week: tuple[int, ...] | None = ...,
+    ) -> None: ...
+    @overload
+    def dates(
+        self,
+        start_date: (
+            np.integer | float | str | _date | datetime | np.datetime64 | None
+        ),
+        end_date: (np.integer | float | str | _date | datetime | np.datetime64 | None),
+        return_name: Literal[False] = ...,
+    ) -> DatetimeIndex: ...
+    @overload
+    def dates(
+        self,
+        start_date: (
+            np.integer | float | str | _date | datetime | np.datetime64 | None
+        ),
+        end_date: (np.integer | float | str | _date | datetime | np.datetime64 | None),
+        return_name: Literal[True] = ...,
+    ) -> Series: ...
+
+holiday_calendars: dict[str, type["AbstractHolidayCalendar"]]
+
+def register(cls: type["AbstractHolidayCalendar"]) -> None: ...
+def get_calendar(name: str) -> "AbstractHolidayCalendar": ...
+
+class AbstractHolidayCalendar:
+    rules: list[Holiday] = ...
+    start_date: Timestamp = ...
+    end_date: Timestamp = ...
+    _cache: list[tuple[Timestamp, Timestamp, Series[datetime]]] | None = ...
+
+    def __init__(self, name: str = "", rules=None) -> None: ...
+    def rule_from_name(self, name: str) -> Holiday | None: ...
+    @overload
+    def holidays(
+        self,
+        start: datetime | None = ...,
+        end: datetime | None = ...,
+        return_name: Literal[False] = False,
+    ) -> DatetimeIndex: ...
+    @overload
+    def holidays(
+        self,
+        start: datetime | None = ...,
+        end: datetime | None = ...,
+        return_name: Literal[True] = ...,
+    ) -> Series[...]: ...
+    @staticmethod
+    def merge_class(
+        base: "AbstractHolidayCalendar"
+        | type["AbstractHolidayCalendar"]
+        | list[Holiday],
+        other: "AbstractHolidayCalendar"
+        | type["AbstractHolidayCalendar"]
+        | list[Holiday],
+    ) -> list[Holiday]: ...
+    @overload
+    def merge(
+        self,
+        other: "AbstractHolidayCalendar" | type["AbstractHolidayCalendar"],
+        inplace: Literal[False] = ...,
+    ) -> list[Holiday]: ...
+    @overload
+    def merge(
+        self,
+        other: "AbstractHolidayCalendar" | type["AbstractHolidayCalendar"],
+        inplace: Literal[True] = ...,
+    ) -> None: ...
```

Comment:
Same as above - swap the overloads.  First one should be `inplace: Literal[True]`, returning `None`.  Second one is `inplace: bool = ...` returning `list[Holiday]`